src/battle/other_battle_calculators.c

```diff
@@ -2605,3 +2615,163 @@ void BattleControllerPlayer_UpdateMonCondition(void *bw, struct BattleStruct *sp
+    
+    if (ctx->oneSelfFlag[ctx->attack_client].trickroom_flag) {
+        SortExecutionOrderBySpeed(bsys, ctx);
+        SortMonsBySpeed(bsys, ctx);
+        ctx->executionIndex = 0;
+    } else {
+        ctx->executionIndex++;
+    }
+    
+    BattleStructureInit(ctx);
+    
+    ctx->server_seq_no = CONTROLLER_COMMAND_8;
+}
+
+/**
+ *  @brief checks if the given move should be weakened or not (only prints message)
+ *  @param bw battle work structure
+ *  @param sp global battle structure
+ *  @return TRUE/FALSE
+ */
+BOOL CheckStrongWindsWeaken(struct BattleSystem *bw, struct BattleStruct *sp) {
+    u8 defender_type_1 = BattlePokemonParamGet(sp, sp->defence_client, BATTLE_MON_DATA_TYPE1, NULL);
+    u8 defender_type_2 = BattlePokemonParamGet(sp, sp->defence_client, BATTLE_MON_DATA_TYPE2, NULL);
+    u32 move_type = GetAdjustedMoveType(sp, sp->attack_client, sp->current_move_index);
+    u8 i = 0;
+    
+    // TODO: Check type3
+    while (TypeEffectivenessTable[i][0] != 0xff) {
+        if (TypeEffectivenessTable[i][0] == move_type) {
+            if ((TypeEffectivenessTable[i][1] == defender_type_1) || (TypeEffectivenessTable[i][1] == defender_type_2)) {
+                if ((!CheckSideAbility(bw, sp, CHECK_ABILITY_ALL_HP, 0, ABILITY_CLOUD_NINE) && !CheckSideAbility(bw, sp, CHECK_ABILITY_ALL_HP, 0, ABILITY_AIR_LOCK)
+                && sp->field_condition & WEATHER_STRONG_WINDS
+                && (TypeEffectivenessTable[i][2] == 20)
+                && ((defender_type_1 == TYPE_FLYING) || (defender_type_2 == TYPE_FLYING)))) {
+                    LoadBattleSubSeqScript(sp, ARC_BATTLE_SUB_SEQ, SUB_SEQ_WEAKEN_MOVES_STRONG_WINDS);
+                    sp->next_server_seq_no = sp->server_seq_no;
+                    sp->server_seq_no = CONTROLLER_COMMAND_RUN_SCRIPT;
+                    return TRUE;
+                }
+            }
+        }
+        i++;
+    }
+
+    return FALSE;
+}
+
+enum {
+    TRY_MOVE_START = 0,
+
+    TRY_MOVE_STATE_CHECK_VALID_TARGET = TRY_MOVE_START,
+    TRY_MOVE_STATE_TRIGGER_REDIRECTION_ABILITIES,
+    TRY_MOVE_STATE_CHECK_MOVE_HITS,
+    TRY_MOVE_STATE_CHECK_MOVE_HIT_OVERRIDES,
+    TRY_MOVE_STATE_CHECK_TYPE_CHART,
+    TRY_MOVE_STATE_TRIGGER_IMMUNITY_ABILITIES,
+    TRY_MOVE_STATE_TRIGGER_STRONG_WINDS,
+
+    TRY_MOVE_END,
+};
+
+/**
+ * Platinum version as reference
+ * BattleController_TryMove
+ * https://github.com/pret/pokeplatinum/blob/04d9ea4cfad3963feafecf3eb0f4adcbc7aa5063/src/battle/battle_controller.c#L3240
+ */
+void LONG_CALL ov12_0224C4D8(struct BattleSystem *bsys, struct BattleStruct *ctx) {
+    // u8 buf[64];
+    // sprintf(buf, "In BattleController_TryMove\n");
+    // debugsyscall(buf);
+
+    switch (ctx->woc_seq_no) {
+    case TRY_MOVE_STATE_CHECK_VALID_TARGET:
+        ctx->woc_seq_no++;
+        if (ov12_0224B398(bsys, ctx) == TRUE) {
+            return;
+        }
+        //fallthrough
```

Comment:
if you put `FALLTHROUGH` here (macro for `__attribute__((fallthrough))`) it will suppress the warnings

Reply:
huh, interesting that the comment does the same thing haha